service/controller/chart/v1/resource/release/desired.go

```diff
@@ -40,41 +42,59 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	configMapValues, err := r.getConfigMapValues(ctx, cr)
+	configMapData, err := r.getConfigMapData(ctx, cr)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	secretValues, err := r.getSecretValues(ctx, cr)
+	secretData, err := r.getSecretData(ctx, cr)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	values, err := union(configMapValues, secretValues)
+	// Merge configmap and secret to provide a single set of values to Helm.
+	values, err := helmclient.MergeValues(configMapData, secretData)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
+	var valuesYAML []byte
+	var valuesMD5Checksum string
+
+	if len(values) > 0 {
+		valuesYAML, err = yaml.Marshal(values)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		valuesMD5Checksum = fmt.Sprintf("%x", md5.Sum(valuesYAML))
+	} else {
+		// We need to pass empty values in ValueOverrides to make the install
+		// process use the default values and prevent errors on nested values.
+		valuesYAML = []byte("{}")
```

Comment:
We still need to pass empty values if no values have been specified. This is so Helm uses the `values.yaml` for the chart.